codegen/src/main/scala/overflowdb/schema/Schema.scala

```diff
@@ -49,6 +49,9 @@ abstract class AbstractNodeType(val name: String, val comment: Option[String], v
   /** all node types that extend this node */
   def subtypes(allNodes: Set[AbstractNodeType]): Set[AbstractNodeType]
 
+  /** the name for the generated node starter. Generatiion of starters can be suppressed by setting to None, or custom
+   * names can be assigned to prevent compile errors for e.g. `type`*/
+  var starterName: Option[String] = Some(camelCase(name))
 
```

Comment:
setter for fluid api, also no need for the caller to wrap it in an option

```suggestion

  def withStarterName(name: String): this.type = {
    this.starterName = Option(name)
    this
  }
  
```

Reply:
:+1: to the fluid API.

Regarding the Option -- I want to give us the possibility of suppressing generation of nodestarters in cases where we don't want them. E.g. TAG_NODE_PAIR could be a candidate. Do you prefer to encode "suppress" via nullpointer or via option-wrapping / None?

Reply:
by default they are disabled, so you're talking about the case where it's defined in a base schema, e.g. `codepropertygraph` and then we want to disable it in a derived schema, e.g. `codescience-schema`?
technically you can pass `null` to `withStarterName`, but that's not really scala style - i'd rather make it explicit&expressive and also add 
```scala
  def withoutStarterName(): this.type = {
    this.starterName = None
    this
  }
```